src/Persistence.Tests/Yaml/DeserializerInvalidTests.cs

```diff
@@ -45,4 +46,35 @@ public void Deserialize_ShouldFailWhenYamlIsInvalid()
         failedFiles.Should().BeGreaterThan(0);
         failedFiles.Should().Be(files.Length, "all files should fail");
     }
+
+    [TestMethod]
+    public void Deserialize_ShouldFailWhenExpectingDifferentType()
+    {
+        // Arrange
+        var deserializer = TestBase.ServiceProvider.GetRequiredService<IYamlSerializationFactory>().CreateDeserializer();
+        using var yamlStream = File.OpenRead("_TestData\\ValidYaml\\Screen\\with-name.pa.yaml");
```

Comment:
nit: use `@` for path names

Reply:
Actually had to swap to `/` like the other tests, as that works for both the Windows and Linux builds